src/DOMParser.ts

```diff
@@ -111,22 +100,43 @@ export class DOMParser implements globalThis.DOMParser {
 
     parser.onopentag = function(node: sax.QualifiedTag) {
       var element = doc.createElementNS(node.uri, node.name) as XMLElement;
-      element.openStart = getPosition(parser.startTagPosition - 1);
-      element.openEnd = { line: parser.line, column: parser.column };
+      element.openStart = parser.startTagPosition - 1;
+      element.nameStart = element.openStart + 1;
+      element.nameEnd = element.nameStart + node.name.length;
+      element.openEnd = parser.position;
+      element.attributePositions = {};
 
       // We have: parser.line, parser.column, parser.position, parser.startTagPosition
       for (var key in node.attributes) {
-        var attr = node.attributes[key];
-        element.setAttributeNS(attr.uri, attr.name, attr.value);
+        var attr = node.attributes[key] as sax.Attribute;
+        // Attribute nodes seem to be deprecated in general.
+        // doc.createAttributeNS does not work for some cases in PhantomJS, e.g.
+        //   `xmlns="http://www.w3.org/2001/XMLSchema"`:
+        //   {name: 'xmlns', value: 'http://www.w3.org/2001/XMLSchema', prefix: 'xmlns', local: '', uri: 'http://www.w3.org/2000/xmlns/'
+        // Also, it was removed in Chrome 34.
+        // Custom attributes on nodes also disappear sometimes in Chrome.
+        // The solution is to store the attribute positions on the element node.
+        var position: XMLAttributePosition = {
+          start: (attr as any).start - 1,
+          end: (attr as any).end,
+          nameEnd: 0,
+          valueStart: 0
+        };
+        position.nameEnd =
+          position.start + (attr.name == null ? 0 : attr.name.length);
+        position.valueStart = position.nameEnd + 1; // NOT always correct
```

Comment:
if not always correct, what do?

Reply:
I think it's in weird cases with additional spacing like `attr= "value"`. Old issue, can fix at another time.